app/assets/javascripts/components/timeline/timeline.jsx

```diff
@@ -271,7 +273,71 @@ const Timeline = createReactClass({
       );
       return i += 1;
     }
-    );
+    ); */
+
+    // Using allWeeks
+    this.props.allWeeks.forEach((week, index) => {
+      const weekAnchorName = `week-${index + 1 + weeksBeforeTimeline}`;
+      // if week is empty
+      if (week.empty) {
+        const emptyWeekKey = `empty-week-${index}`;
+        weekNavInfo.push({ emptyWeek: true, title: undefined });
+        weekComponents.push((
+          <div key={emptyWeekKey}>
+            <a className="timeline__anchor" name={weekAnchorName} />
+            <EmptyWeek
+              course={this.props.course}
+              edit_permissions={this.props.edit_permissions}
+              index={index + 1}
+              usingCustomTitles={usingCustomTitles}
+              timeline_start={this.props.course.timeline_start}
+              timeline_end={this.props.course.timeline_end}
+              weeksBeforeTimeline={weeksBeforeTimeline}
+              addWeek={this.props.addWeek}
+            />
+          </div>
+        ));
+      } else {
+        weekNavInfo.push({ emptyWeek: false, title: week.title });
+        weekComponents.push((
+          <div key={week.id}>
+            <a className="timeline__anchor" name={weekAnchorName} />
+            <Week
+              week={week}
+              index={index + 1}
+              reorderable={this.props.reorderable}
+              editableTitles={this.props.editableTitles}
+              usingCustomTitles={usingCustomTitles}
+              updateTitle={this.props.updateTitle}
+              blocks={week.blocks}
+              deleteWeek={this.deleteWeek.bind(this, week.id)}
+              meetings={week.meetings}
+              timeline_start={this.props.course.timeline_start}
+              timeline_end={this.props.course.timeline_end}
+              all_training_modules={this.props.all_training_modules}
+              editableBlockIds={this.props.editableBlockIds}
+              edit_permissions={this.props.edit_permissions}
+              saveBlockChanges={this.props.saveBlockChanges}
+              setBlockEditable={this.props.setBlockEditable}
+              cancelBlockEditable={this.props.cancelBlockEditable}
+              updateBlock={this.props.updateBlock}
+              addBlock={this.props.addBlock}
+              deleteBlock={this.props.deleteBlock}
+              saveGlobalChanges={this.props.saveGlobalChanges}
+              canBlockMoveUp={this._canBlockMoveUp.bind(this, week, (week.order - 1))}
```

Comment:
May be, there is something wrong which I have done here. But, I can't figure it out.